sharing/resource_share.go

```diff
@@ -29,13 +29,13 @@ const (
 type ShareInfo struct {
 	Name      string             `json:"name" tf:"force_new"`
 	Owner     string             `json:"owner,omitempty" tf:"suppress_diff"`
-	Objects   []SharedDataObject `json:"objects,omitempty" tf:"alias:object"`
+	Objects   []SharedDataObject `json:"objects,omitempty" tf:"slice_set,alias:object"`
 	CreatedAt int64              `json:"created_at,omitempty" tf:"computed"`
 	CreatedBy string             `json:"created_by,omitempty" tf:"computed"`
 }
 
 type SharedDataObject struct {
-	Name                     string      `json:"name"`
+	Name                     string      `json:"name" tf:"force_new"`
```

Comment:
As discussed, let's see if this works without force_new.

Reply:
If we remove force_new, then we need to check for element with empty Name (when we remove an object). This happens during dataToStruct. 

`````
if afterSdo.Name == "" {
    continue
}
`````

Integration test passes, I also added more steps. The comment gets updated. I need to check if this gets shown in diff.

Reply:
Diff doesn't get shown when we update the comment, although the changes are reflected in the request we make and get from server. 

Diff only works for add/delete/rename

Reply:
Update after discussion: https://github.com/databricks/terraform-provider-databricks/pull/3540.

Reply:
> Diff doesn't get shown when we update the comment, although the changes are reflected in the request we make and get from server.
> 
> Diff only works for add/delete/rename

I realized this code doesn't allow updating the Share objects comment. Added that in #3629